src/components/NavigationBar.jsx

```diff
@@ -52,9 +65,11 @@ class NavigationBar extends Component {
 
   render() {
     const {
-      username, userID, numNotifications, keyword,
+      username, numNotifications, keyword, loading,
     } = this.state;
+    const { user } = this.props;
     return (
+      !loading && (
```

Comment:
I've never seen this before so just wanted to confirm. Is this the same as doing:

```
if (!loading) {
    # return navbar code
} else {
    # return nothing
}

Reply:
yes. If we return it directly the count badge will flash in after the navbar has been shown, it doesn't look good.